src/OpenTelemetry.Instrumentation.EventCounters/EventCounterListener.cs

```diff
@@ -0,0 +1,245 @@
+                };
+
+                this.EnableEvents(source, EventLevel.Verbose, EventKeywords.All, arguments);
+            }
+            catch (Exception ex)
+            {
+                EventCountersInstrumentationEventSource.Log.ErrorEventCounter(source.Name, ex.Message);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            if (!this.isInitialized || !eventData.EventName.Equals("EventCounters"))
+            {
+                return;
+            }
+
+            try
+            {
+                if (eventData.Payload.Count > 0 && eventData.Payload[0] is IDictionary<string, object> eventPayload)
+                {
+                    this.ExtractAndPostMetric(eventData.EventSource.Name, eventPayload);
+                }
+                else
+                {
+                    EventCountersInstrumentationEventSource.Log.IgnoreEventWrittenAsEventPayloadNotParseable(eventData.EventSource.Name);
+                }
+            }
+            catch (Exception ex)
+            {
+                EventCountersInstrumentationEventSource.Log.ErrorEventCounter(eventData.EventName, ex.ToString());
+            }
+        }
+
+        private void ExtractAndPostMetric(string eventSourceName, IDictionary<string, object> eventPayload)
+        {
+            try
+            {
+                bool calculateRate = false;
+                double actualValue = 0.0;
+                double actualInterval = 0.0;
+                double recordedValue = 0.0;
+
+                string counterName = string.Empty;
+                string counterDisplayName = string.Empty;
+                InstrumentType instrumentType = InstrumentType.Counter;
+
+                foreach (KeyValuePair<string, object> payload in eventPayload)
+                {
+                    var key = payload.Key;
+                    if (key.Equals("Name", StringComparison.OrdinalIgnoreCase))
+                    {
+                        counterName = payload.Value.ToString();
+                    }
+                    else
+                    if (key.Equals("DisplayName", StringComparison.OrdinalIgnoreCase))
+                    {
+                        counterDisplayName = payload.Value.ToString();
+                    }
+                    else if (key.Equals("Mean", StringComparison.OrdinalIgnoreCase))
+                    {
+                        instrumentType = InstrumentType.Counter;
+                    }
+                    else if (key.Equals("Increment", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Increment indicates we have to calculate rate.
+                        instrumentType = InstrumentType.Gauge;
+                        calculateRate = true;
+                    }
+                    else if (key.Equals("IntervalSec", StringComparison.OrdinalIgnoreCase))
```

Comment:
I'd suggest to ignore this whole thing. lets just trust that EventCounter will fire based on our configuration, and we always assume it is correct.

Reply:
it should simple this whole class a lot!

Reply:
You mean just the section that reads and verifies actual interval or just the whole calculation of recordedValue based on rate?

Reply:
the part which reads "actual" interval. If we simply assume its always what we configured, it'll simplify a lot.
(I dont recollect if there is a reason for us to *not trust* eventcounters to not respect the configured interval.)

(Will ask the eventcounter owners to chime in here, if needed)

Reply:
I made this change to remove the logic to read actual interval.